flang/lib/Lower/OpenMP.cpp

```diff
@@ -2430,6 +2451,75 @@ genEnterExitDataOp(Fortran::lower::AbstractConverter &converter,
                                    deviceOperand, nowaitAttr, mapOperands);
 }
 
+static void genBodyOfTargetOp(
```

Comment:
This function has to be commented more since all the variables that are accessed in the region are made block arguments here and this helps implementing the `IsolatedFromAbove` Trait.